lib/Viroverse/ISLAWorksheet.pm

```diff
@@ -0,0 +1,260 @@
+use strict;
+use warnings;
+use utf8;
+
+package Viroverse::ISLAWorksheet;
+use Moo;
+use Types::Standard qw< :types >;
+use Viroverse::Logger qw< :log >;
+
+=head1 NAME
+
+Viroverse::ISLAWorksheet - convert between a Viroverse::Result::Sample and an
+Excel spreadsheet for ISLA data
+
+=head1 ATTRIBUTES
+
+=head2 model
+
+The Viroverse::Model::sample instance that this spreadsheet represents
+
+=head1 METHODS
+
+=head2 make_xlsx
+
+Generates an Excel spreadsheet (in xlsx format) that can be filled out with
+added data for the sample. Returns a FileHandle that has been seeked to the
+beginning.
+
+=cut
+
+has model => (
+    is       => 'ro',
+    isa      => Object,
```

Comment:
Use `ViroDBRecord["Sample"]` (from `Viroverse::Types`) as the type parameter here.